src/dbnode/storage/bootstrap/bootstrapper/commitlog/source.go

```diff
@@ -831,50 +929,40 @@ func (s *commitLogSource) bootstrapShardBlockSnapshot(
 		zap.Time("blockStart", blockStart),
 		zap.Int("volume", mostRecentCompleteSnapshot.ID.VolumeIndex))
 
-	for {
-		id, tags, data, expectedChecksum, err := reader.Read()
-		if err != nil && err != io.EOF {
-			return err
-		}
-		if err == io.EOF {
-			break
-		}
+	worker := &readSeriesBlocksWorker{
+		dataCh:      make(chan seriesBlock, readSeriesBlocksWorkerChannelSize),
+		reader:      reader,
+		shard:       shard,
+		accumulator: accumulator,
+		blocksPool:  blocksPool,
+		blockStart:  blockStart,
+		blockSize:   blockSize,
+		nsCtx:       nsCtx,
+	}
 
-		dbBlock := blocksPool.Get()
-		dbBlock.Reset(blockStart, blockSize,
-			ts.NewSegment(data, nil, 0, ts.FinalizeHead), nsCtx)
+	ctx := context.NewBackground()
+	errs, _ := errgroup.WithContext(ctx.GoContext())
+	errs.Go(worker.readSeriesBlocks)
+	if err := s.loadBlocks(worker.dataCh, writeType); err != nil {
```

Comment:
I think it makes sense to do this. I think I'll reuse `AccumulateConcurrency` config property because it is essentially describes what we are doing and also introducing additional config for this task is probably too much.